pkg/utils/types.go

```diff
@@ -95,40 +97,67 @@ const (
 
 // GenerateClientSetFromKubeConfig will generation both ClientSets (k8s, and Litmus)
 func (clientSets *ClientSets) GenerateClientSetFromKubeConfig() error {
-	config, err := getKubeConfig()
+	configExperiment, configLitmus, err := getKubeConfig()
 	if err != nil {
 		return err
 	}
-	k8sClientSet, err := k8s.GenerateK8sClientSet(config)
+
+	k8sClientSetLitmus, err := k8s.GenerateK8sClientSet(configLitmus)
 	if err != nil {
 		return err
 	}
-	litmusClientSet, err := litmus.GenerateLitmusClientSet(config)
+	litmusClientSetLitmus, err := litmus.GenerateLitmusClientSet(configLitmus)
+	if err != nil {
+		return err
+	}
+
+	k8sClientSetExperiment, err := k8s.GenerateK8sClientSet(configExperiment)
 	if err != nil {
 		return err
 	}
-	clientSets.KubeClient = k8sClientSet
-	clientSets.LitmusClient = litmusClientSet
+	litmusClientSetExperiment, err := litmus.GenerateLitmusClientSet(configExperiment)
+	if err != nil {
+		return err
+	}
+
+	clientSets.KubeClient = k8sClientSetLitmus
+	clientSets.LitmusClient = litmusClientSetLitmus
+	clientSets.KubeClientExperiment = k8sClientSetExperiment
+	clientSets.LitmusClientExperiment = litmusClientSetExperiment
 
 	return nil
 }
 
 // Generate
 
 // getKubeConfig setup the config for access cluster resource
-func getKubeConfig() (*rest.Config, error) {
+func getKubeConfig() (*rest.Config, *rest.Config, error) {
 	kubeconfig := flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	litmuskubeconfig := flag.String("litmuskubeconfig", "", "absolute path to the kubeconfig file")
 	flag.Parse()
+	// Use in-cluster config if kubeconfig path is specified
+	if *litmuskubeconfig == "" {
+		configLitmus, err := rest.InClusterConfig()
```

Comment:
Should this not be configExperiment ?

Reply:
Actually no, whatever is specified by `litmuskubeconfig` parameter is then defined in `configLitmus` variable. And whatever is specified in `kubeconfig` parameter is then defined in `configExperiment` variable.

Reply:
Ok got it. Could we name this better. Can we say `litmusControlPlaneKubeConfig ` as the one which deals w/ chaos-operator/runner/experimentCR/engineCR & `targetKubeConfig` as the one that deals w/ the target application & also the  experiment job/helper pod & chaosresult resources (note that chaosresult CR is created by the experiment and thereby will reside in the target cluster)

Reply:
Will do.